static/app/views/dashboards/widgets/timeSeriesWidget/timeSeriesWidgetVisualization.tsx

```diff
@@ -164,38 +166,85 @@ export function TimeSeriesWidgetVisualization(props: TimeSeriesWidgetVisualizati
+
+  // Count up the field types of all the plottables
+  const fieldTypeCounts = mapValues(plottablesByType, plottables => plottables.length);
+
+  // Sort the field types by how many plottables use each one
+  const axisTypes = (Object.keys(fieldTypeCounts) as AggregationOutputType[])
+    .toSorted(
+      // `dataTypes` is extracted from `dataTypeCounts`, so the counts are guaranteed to exist
+      (a, b) => fieldTypeCounts[b]! - fieldTypeCounts[a]!
+    )
+    .filter(axisType => !!axisType); // `TimeSeries` allows for a `null` data type , though it's not likely
+
+  // Assign most popular field type to left axis
+  const leftYAxisType = axisTypes.at(0) ?? FALLBACK_TYPE;
+
+  // Assign the rest of the field types to right
+  const rightYAxisTypes = axisTypes.slice(1);
+
+  // Narrow down to just one type for the right Y axis If there's just one field
+  // type for the right axis, use it. If there are more than one, fall back to a
+  // default. If there are 0, there's no type for the right axis, and we won't
+  // plot one
+  const rightAxisType =
+    rightYAxisTypes.length === 1
+      ? rightYAxisTypes[0]!
+      : rightYAxisTypes.length > 2
+        ? FALLBACK_TYPE
+        : undefined;
+
+  // Create a map of used units by plottable data type
+  const unitsByType = mapValues(plottablesByType, plottables =>
+    uniq(plottables.map(plottable => plottable.dataUnit))
+  );
 
-  let yAxisUnit: DurationUnit | SizeUnit | RateUnit | null;
+  // Narrow down to just one unit for each plottable data type
+  const unitForType = mapValues(unitsByType, (relevantUnits, type) => {
+    if (relevantUnits.length === 1) {
+      // All plottables of this type have the same unit
+      return relevantUnits[0]!;
+    }
 
-  // N.B. Do not filter `boolean` because `null` is a valid unit
-  const units = uniq(props.plottables.map(plottable => plottable.dataUnit));
+    if (relevantUnits.length === 0) {
+      // None of the plottables of this type supplied a unit
+      return FALLBACK_UNIT_FOR_FIELD_TYPE[type as AggregationOutputType];
+    }
 
-  if (units.length === 1) {
-    // All timeseries have the same unit. Use that unit. This is especially
-    // important for named rate timeseries like `"epm()"` where the user would
-    // expect a plot in minutes
-    yAxisUnit = units[0]!;
-  } else {
-    // None of the series specified a unit, or there are mismatched units. Fall
-    // back to an appropriate unit for the axis type
-    yAxisUnit = FALLBACK_UNIT_FOR_FIELD_TYPE[yAxisFieldType];
-  }
+    if (relevantUnits.length > 1) {
+      // Plottables of this type has mismatched units. Return a fallback. It
+      // would also be acceptable to return the unit of the _first_ plottable,
+      // probably
+      return FALLBACK_UNIT_FOR_FIELD_TYPE[type as AggregationOutputType];
+    }
+
+    // This return exists only to appease TypeScript
+    return FALLBACK_UNIT_FOR_FIELD_TYPE[type as AggregationOutputType];
```

Comment:
I think you can drop the last conditional since you cover both scenarios above, and this is the same `else` clause if you cover these to `if/else if/else`

Reply:
Roger, dropped